packages/react-components/src/components/RemoteVideoTile.tsx

```diff
@@ -45,25 +45,35 @@ export const RemoteVideoTile = React.memo(
       if (isAvailable && !renderElement) {
         onCreateRemoteStreamView && onCreateRemoteStreamView(userId, remoteVideoViewOptions);
       }
-      if (!isAvailable) {
-        onDisposeRemoteStreamView && onDisposeRemoteStreamView(userId);
-      }
+      // Always clean up element to make tile up to date and be able to dispose correctly
+      // TODO: Add an extra param to onDisposeRemoteStreamView(userId, flavor(optional)) when api lock is released
+      // and isolate dispose behavior between screen share and video
+      return () => {
+        if (renderElement && !isScreenSharingOn) {
```

Comment:
I'm jittery with this logic on two accounts.

1. What happens if `isScreenSharingOn` is set after this component mounts. In particular, consider -> mount (useEffect calls onCreateRemoteStreamView) -> isScreenSharingOn is set to true -> useEffect cleanup is noop -> useEffect *again* calls onCreateRemoteStreamView -> renderElement gets populated -> useEffect cleanup is noop -> useEffect is noop -> a noop useEffect is registered (because `isScreenSharingOn`). In this case, this component called `onCreateRemoteStreamView` twice, and will not cleanup ever. The ScreenShareTile _should_ cleanup.
2. The cleanup registered in `useEffect` below also somewhat races with this useEffect. In the example sequence of events above, when `isScreenSharingOn` changes, the cleanup for that useEffect will run, and call onDisposeRemoteStreamView.... 

In general, it's scary when `isScreenSharingOn` changes with this component already mounted.

Reply:
If the isScreenSharingOn is changed in that way, the screenShare will be disposed first, then it will cause the re-render of screenShareTile, then screenShare tile will notice renderElement is undefined, then it will create something for itself - it happens very rare, but once we have isolated function for screenShare and videoStream, we won't need this strange logic anymore